tests/integration/test_inout.py

```diff
@@ -137,3 +137,40 @@ def test(q1: qubit, q2: qubit, n: int) -> tuple[qubit, qubit]:
 
     validate(module.compile())
 
+
+def test_tensor(validate):
+    module = GuppyModule("test")
+    module.load(quantum)
+
+    @guppy.struct(module)
+    class A:
+        q: qubit
+
+    @guppy.struct(module)
+    class B:
+        q: qubit
+        x: int
+
+    @guppy.struct(module)
+    class C:
+        q: qubit
+        x: float
+
+    @guppy.declare(module)
+    def foo(a: A @ inout, x: int) -> None: ...
+
+    @guppy.declare(module)
+    def bar(y: float, b: B @ inout, c: C) -> C: ...
+
+    @guppy.declare(module)
+    def baz(c: C @ inout) -> None: ...
+
+    @guppy(module)
+    def test(a: A, b: B, c1: C, c2: C, x: bool) -> tuple[A, B, C, C]:
+        c1 = (foo, bar, baz)(a, b.x, c1.x, b, c1, c2)
+        if x:
+            c1 = ((foo, bar), baz)(a, b.x, c1.x, b, c1, c2)
+        c1 = (foo, (bar, baz))(a, b.x, c1.x, b, c1, c2)
```

Comment:
That's some weird syntax for a user-facing language.
I hope it will be documented somewhere?

Reply:
Opened a discussion: https://github.com/CQCL/guppylang/discussions/357